src/ActivityLogger.php

```diff
@@ -2,165 +2,161 @@
 
 namespace Spatie\Activitylog;
 
-use Illuminate\Auth\AuthManager;
+use Closure;
+use DateTimeInterface;
 use Illuminate\Contracts\Config\Repository;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Carbon;
 use Illuminate\Support\Str;
 use Illuminate\Support\Traits\Macroable;
 use Spatie\Activitylog\Contracts\Activity as ActivityContract;
-use Spatie\Activitylog\Exceptions\CouldNotLogActivity;
 
 class ActivityLogger
 {
     use Macroable;
 
-    /** @var \Illuminate\Auth\AuthManager */
-    protected $auth;
+    protected ?string $defaultLogName = null;
 
-    protected $defaultLogName = '';
+    protected CauserResolver $causerResolver;
 
-    /** @var string */
-    protected $authDriver;
+    protected ActivityLogStatus $logStatus;
 
-    /** @var \Spatie\Activitylog\ActivityLogStatus */
-    protected $logStatus;
+    protected ?ActivityContract $activity = null;
 
-    /** @var \Spatie\Activitylog\Contracts\Activity */
-    protected $activity;
+    protected LogBatch $batch;
 
-    public function __construct(AuthManager $auth, Repository $config, ActivityLogStatus $logStatus)
+    public function __construct(Repository $config, ActivityLogStatus $logStatus, LogBatch $batch, CauserResolver $causerResolver)
     {
-        $this->auth = $auth;
+        $this->causerResolver = $causerResolver;
 
-        $this->authDriver = $config['activitylog']['default_auth_driver'] ?? $auth->getDefaultDriver();
+        $this->batch = $batch;
 
         $this->defaultLogName = $config['activitylog']['default_log_name'];
 
         $this->logStatus = $logStatus;
     }
 
-    public function setLogStatus(ActivityLogStatus $logStatus)
+    public function setLogStatus(ActivityLogStatus $logStatus): static
     {
         $this->logStatus = $logStatus;
 
         return $this;
     }
 
-    public function performedOn(Model $model)
+    public function performedOn(Model $model): static
     {
         $this->getActivity()->subject()->associate($model);
 
         return $this;
     }
 
-    public function on(Model $model)
+    public function on(Model $model): static
     {
         return $this->performedOn($model);
     }
 
-    public function causedBy($modelOrId)
+    public function causedBy(Model | int | null $modelOrId): static
```

Comment:
We should allow `UUID` as well. The migration isn't compatible by default but that can be adjusted by user - this part can't.

```suggestion
    public function causedBy(Model | int | string | null $modelOrId): static
```